common/src/main/java/org/opensearch/ml/common/input/nlp/TextSimilarityMLInput.java

```diff
@@ -0,0 +1,116 @@
+import org.opensearch.ml.common.input.MLInput;
+
+import static org.opensearch.core.xcontent.XContentParserUtils.ensureExpectedToken;
+
+
+/**
+ * MLInput which supports a text similarity algorithm
+ * Inputs are a query and a list of texts. Outputs are real numbers
+ * Use this for Cross Encoder models
+ */
+@org.opensearch.ml.common.annotation.MLInput(functionNames = {FunctionName.TEXT_SIMILARITY})
+public class TextSimilarityMLInput extends MLInput {
+
+    public TextSimilarityMLInput(FunctionName algorithm, MLInputDataset dataset) {
+        super(algorithm, null, dataset);
+    }
+
+    public TextSimilarityMLInput(StreamInput in) throws IOException {
+        super(in);
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        super.writeTo(out);
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        builder.field(ALGORITHM_FIELD, algorithm.name());
+        if(parameters != null) {
+            builder.field(ML_PARAMETERS_FIELD, parameters);
+        }
+        if(inputDataset != null) {
+            TextSimilarityInputDataSet ds = (TextSimilarityInputDataSet) this.inputDataset;
+            List<String> docs = ds.getTextDocs();
+            String queryText = ds.getQueryText();
+            builder.field(QUERY_TEXT_FIELD, queryText);
+            if (docs != null && !docs.isEmpty()) {
+                builder.startArray(TEXT_DOCS_FIELD);
+                for(String d : docs) {
+                    builder.value(d);
+                }
+                builder.endArray();
+            }
+        }
+        builder.endObject();
+        return builder;
+    }
+
+    public TextSimilarityMLInput(XContentParser parser, FunctionName functionName) throws IOException {
+        super();
+        this.algorithm = functionName;
+        List<String> docs = new ArrayList<>();
+        String queryText = null;
+
+        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.currentToken(), parser);
+        while (parser.nextToken() != XContentParser.Token.END_OBJECT) {
+            String fieldName = parser.currentName();
+            parser.nextToken();
+
+            switch (fieldName) {
+                case TEXT_DOCS_FIELD:
+                    ensureExpectedToken(XContentParser.Token.START_ARRAY, parser.currentToken(), parser);
+                    while (parser.nextToken() != XContentParser.Token.END_ARRAY) {
+                        String context = parser.text();
+                        docs.add(context);
+                    }
+                    break;
+                case QUERY_TEXT_FIELD: 
+                    queryText = parser.text();
+                default:
+                    parser.skipChildren();
+                    break;
+            }
+        }        
+        if(docs.isEmpty()) {
+            throw new IllegalArgumentException("no text docs");
```

Comment:
[nit] what do you think about: `No text documents are provided`

Reply:
ack